### PR TITLE
Adds high-level Python client documentation

### LINKED ISSUE #1090: [DOC]Add documentation for the high-level Python client
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Add documentation for the high-level Python client

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #1090 

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by harshavamsi — APPROVED** [review]
LGTM!

**Review by cwillum — APPROVED** [review]
Looks great. Just one comment.

File: _clients/python-high-level.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+```
+
+After installing the client, you can import it like any other module:
+
+```python
+from opensearchpy import OpenSearch
+from opensearch_dsl import Search
+```
+
+If you prefer to add the client manually or just want to examine the source code, see [opensearch-dsl-py on GitHub](https://github.com/opensearch-project/opensearch-dsl-py).
```
Comment (alicejw-aws) [inline _clients/python-high-level.md:26]:
```suggestion
If you prefer to add the client manually or just want to examine the source code, see [opensearch-dsl-py](https://github.com/opensearch-project/opensearch-dsl-py) on GitHub.
```

File: _clients/python-high-level.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    year = Text()
+
+    class Index:
+        name = index_name
+
+    def save(self, ** kwargs):
+        return super(Movie, self).save(** kwargs)
+```
+
+To index a document into OpenSearch, create an object of the new class and call its `save()` method:
```
Comment (alicejw-aws) [inline _clients/python-high-level.md:129]: suggestion: 
"To index a document for use with OpenSearch, "

File: _clients/python-low-level.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    }
+  }
+}
+
+response = client.indices.create(index_name, body=index_body)
+```
+
+## Indexing a document
+
+You can index a document into OpenSearch using the `client.index()` method:
```
Comment (alicejw-aws) [inline _clients/python-low-level.md:113]: suggestion

"You can index a document for use with OpenSearch. . ."

File: _clients/python-low-level.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+```python
+movies = '{ "index" : { "_index" : "my-dsl-index", "_id" : "2" } } \n { "title" : "Interstellar", "director" : "Christopher Nolan", "year" : "2014"} \n { "create" : { "_index" : "my-dsl-index", "_id" : "3" } } \n { "title" : "Star Trek Beyond", "director" : "Justin Lin", "year" : "2015"} \n { "update" : {"_id" : "3", "_index" : "my-dsl-index" } } \n { "doc" : {"year" : "2016"} }'
+
+client.bulk(movies)
+```
+
+## Searching for documents
+
+The easiest way to search for documents is to construct a query string. The following code uses a multi_match query to search for “miller” in the title and director fields. It boosts the documents where “miller” is in the title field:
```
Comment (alicejw-aws) [inline _clients/python-low-level.md:142]: multi_match -> multi match 
or use code font

**Review by alicejw-aws — APPROVED** [review]
Nice work!

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _clients/python-high-level.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+```python
+s = Search(using=client, index=index_name) \
+    .filter("term", year="2011") \
+    .query("match", title="Moneyball")
+
+response = s.execute()
+```
+
+The preceding query is equivalent to the following query in OpenSearch domain-specific language:
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _clients/python-high-level.md:160]: This is the only mention of this term, so do we still have to add the abbreviation?

[the remark this answers is not in the document]
Comment (natebower) [reply _clients/python-high-level.md:160]: Technically, no, but could it help the reader to cross-reference?

**Pull request merged**